apps/extension/src/background.ts

```diff
@@ -24,41 +23,49 @@ chrome.runtime.onMessageExternal.addListener(
     // check if the request contains our expected property
     // REVIEW: Should this be some random encrypted key for added security?
     console.assert(Boolean(request.accessToken))
-    if (!request?.accessToken) {
-      return
-    }
+    if (request?.accessToken) {
+      // get out storage value
+      const storageValue = await storage.get<unknown>("accessToken")
+      // validate it according to our schema
+      const accessToken = ChromeStorage.accessToken.safeParse(storageValue)
+      // skip storage mutation if we already have that token
+      if (accessToken.success && accessToken.data === request.accessToken) {
+        console.log("RESPONDING: ", { success: true })
+        sendResponse({ success: true })
+        return
+      }
 
-    // get out storage value
-    const storageValue = await storage.get<unknown>("accessToken")
-    // validate it according to out schema
-    const accessToken = ChromeStorage.accessToken.parse(storageValue)
-    // skip storage mutation if we already have that token
-    if (accessToken === request.accessToken) {
-      console.log("RESPONDING: ", { success: true })
-      sendResponse({ success: true })
-      return
-    }
+      // update chrome storage w/ token
+      await storage.set("accessToken", request.accessToken)
 
-    // update chrome storage w/ token
-    await storage.set("accessToken", request.accessToken)
-
-    const updatedAccessToken = await storage.get<unknown>("accessToken")
-    console.log("RESPONDING: ", {
-      success: updatedAccessToken === request.accessToken
-    })
-    sendResponse({
-      success: updatedAccessToken === request.accessToken
-    })
+      const updatedAccessToken = await storage.get<unknown>("accessToken")
+      console.log("RESPONDING: ", {
+        success: updatedAccessToken === request.accessToken
+      })
+      sendResponse({
+        success: updatedAccessToken === request.accessToken
+      })
+    } else if (request === "signout") {
+      // reset chrome storage
+      await storage.set("accessToken", null)
+      console.log("RESPONDING: ", {
+        success: (await storage.get<unknown>("accessToken")) === null
+      })
+      // respond with success message
+      sendResponse({
+        success: (await storage.get<unknown>("accessToken")) === null
+      })
+    }
     return
   }
 )
 
 chrome.storage.onChanged.addListener((changes) => {
   if (changes.accessToken) {
-    // invalidate rq cache
-    queryClient.invalidateQueries([changes.accessToken])
+    // FIXME: Why am I using queryClient -- isn't useContext the correct way to invalidate?
```

Comment:
correct, although i'm not sure which query to invalidate, `changes.accessToken` isn't a valid querykey?

Reply:
Correct. We're currently using @plasmo/storage to store the accessToken and not trpc.

I probably wanted to add [trpc-chrome](https://github.com/jlalmes/trpc-chrome) for storage usage instead of plasmo's package but then forgot? Sorry about this, I just left my stashed changes idle for a bit too long.